client/my-sites/plugins/plugin-details-header/index.jsx

```diff
@@ -16,6 +16,21 @@ const PluginDetailsHeader = ( { plugin, isPlaceholder, isJetpackCloud } ) => {
 
 	const selectedSite = useSelector( getSelectedSite );
 
+	// This gets the version of the plugin on the last site or
+	// the wporg version if that doesn't exist. This is preserving
+	// some pre-existing behavior during a refactor and doesn't
+	// necessarily have to remain this way in the future.
+	let version = plugin?.version;
+	if ( plugin.sites ) {
+		const sites = Object.values( plugin.sites );
+		for ( let i = sites.length - 1; i >= 0; i-- ) {
+			if ( sites[ i ].version ) {
+				version = sites[ i ].version;
+				break;
+			}
+		}
+	}
+
```

Comment:
I agree, but I think the behavior needs to be better defined first before a selector that could be re-used is created. Previously the main selector returned a `version` property on the plugin. However, in the data returned from the server the version is given per site. A plugin can be installed on several different sites with a different version on each site. So what does it mean for a plugin itself to have a version?

Previously it was just the version of whatever site was last in the list because the version of each site overwrote the version of the last one. There were several properties like this, things specific to a site that were being lifted to the plugin object and the last site determining what the values are. I removed all of those and left them only on the site objects.

There should probably be selectors for getting the lowest version on a site, highest version on a site, and most recent version (i.e. the wporg version), but I feel that warrants its own PR.

Reply:
A separate PR for such a selector sounds good to me. We can include the priority of versioning that you described in the inline docs of that selector. Some unit tests would also help.